src/runtimes/index.js

```diff
@@ -1,8 +1,82 @@
+const { extname, basename } = require('path')
+
 const goRuntime = require('./go')
 const jsRuntime = require('./node')
 const rustRuntime = require('./rust')
 
+/**
+ * Finds functions for a list of paths using a specific runtime. The return
+ * value is an object containing an array of the functions found (`functions`)
+ * and an array with the paths that haven't been recognized by the runtime
+ * (`remainingPaths`).
+ *
+ * @param   {Array<String>} paths
+ * @param   {Object} runtime
+ * @returns {Promise<Object>}
+ */
+const findFunctionsInRuntime = async function ({ dedupe = false, paths, runtime }) {
+  const functions = await runtime.findFunctionsInPaths(paths)
+
+  // If `dedupe` is true, we use the function name (`filename`) as the map key,
+  // so that `function-1.js` will overwrite `function-1.go`. Otherwise, we use
+  // `srcPath`, so that both functions are returned.
+  const key = dedupe ? 'name' : 'srcPath'
+
+  // Augmenting the function objects with additional information.
+  const augmentedFunctions = functions.map((func) => [
+    func[key],
+    {
+      ...func,
+      extension: extname(func.mainFile),
+      filename: basename(func.srcPath),
+      runtime,
+    },
+  ])
+  const usedPaths = new Set(augmentedFunctions.map(([path]) => path))
+  const remainingPaths = paths.filter((path) => !usedPaths.has(path))
```

Comment:
This will ensure that once a path has been "taken" by a runtime, it will not be sent to the next runtime, since it will add unnecessary computations.